components/DiffView.tsx

```diff
@@ -93,11 +94,29 @@ const DiffView: React.FC<{
     return (
       <div className="position-relative">
         <div className="position-absolute w-100 d-flex justify-content-end p-1">
+          <div className="form-check p-2">
+            <input
+              className="form-check-input"
+              type="checkbox"
+              value=""
+              id={`checkBox-${id}-${fileIdx}`}
```

Comment:
question why do we need `fileIdx` here? 

Looking at just the code, a `uuid` would have worked too right? or `Math.rand()*Date.now()`

Reply:
Yeah that would have worked too. Now looking at the bug that flacial stated below, I used fileIdx to map each of the files viewableStates. I'm not sure if that was a good solution or not. 

I can definitely change the id above to `checkBox-${id}-${Math.rand()*Date.now()}`. 

I thought it would make more sense if it was mapped to the submissionId and then the id of the file for each diff i.e. `checkbox-100-0`, `checkbox-100-1 etc`